backend/de.metas.monitoring/src/main/java/de/metas/monitoring/adapter/PerformanceMonitoringData.java

```diff
@@ -37,11 +37,18 @@
 
 	public String getInitiatorFunctionNameFQ()
 	{
-		return !calledBy.isEmpty() ? calledBy.get(0).getFunctionNameFQ() : "";
+		if (calledBy.isEmpty())
+		{
+			return "";
+		}
+		else
+		{
+			return calledBy.size() > 1 ? calledBy.get(1).getFunctionNameFQ() : calledBy.get(0).getFunctionNameFQ();
```

Comment:
> calledBy.size() > 1

this does not look good to me. It's a hidden feature which could be broken anytime in the future.

I suggest to:
* introduce a flag in PerformanceMonitoringService.Metadata, e.g. `isGrouppingPlaceholder`. A flag which you will set when you will create the dummy "HTTP request" grouping metadata.

* in this method i think u could take the first one which has isGrouppingPlaceholder=false